submissions/SamVal007/Frogger game task/app.js

```diff
@@ -0,0 +1,129 @@
+const playerStartValue = {
+    "x": 200,
+    "y": 400,
+    "icon": "images/char-boy.png",
+    "stepX": 100,
+    "stepY": 82,
+    "startScore": 0
+};
+
+const enemyStartValue = {
+    "x": 0,
+    "icon": "images/enemy-bug.png",
+    "minX": -50,
+    "maxX": 510,
+    "firstEnemy": 63,
+    "secondEnemy": 147,
+    "thirdEnemy": 230
+};
+
+const coordsGameGround = {
+    "min": 0,
+    "max": 400,
+    "delay": 200,
+    "conflictX": 80,
+    "conflictY": 60,
+    "minSpeed": 100,
+    "maxSpeed": 222
+};
+
+class Character {
+    constructor(x, y, sprite) {
+        this.x = x;
+        this.y = y;
+        this.sprite = sprite;
+    }
+
+    render() {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+}
+
+class Enemy extends Character {
+    constructor(x, y, sprite, player) {
+        super(x, y, sprite);
+        this.speed = this.getRandomSpeed();
+        this.player = player;
```

Comment:
`this.player` is used once only.
Consider this as a hint for a problem highlighted in a yet another comment here.

Reply:
I got it!